frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/CassandraRecoveryPlanOverrider.java

```diff
@@ -113,7 +113,48 @@ private Phase getNodeRecoveryPhase(Plan inputPlan, int index) {
                 new UnconstrainedLaunchConstrainer(),
                 stateStore);
 
+        // Restart all other nodes if replacing a seed node to refresh IP resolution
+        List<Step> restartSteps = new ArrayList<>();
+        int replaceIndex = replaceStep.getPodInstanceRequirement().get().getPodInstance().getIndex();
+        if (isSeedNode(replaceIndex)) {
+            logger.info(
+                    "node-{} is a seed node. Restarting all other nodes to refresh seed node address.",
+                    replaceIndex);
+
+            for (Step step : inputPhase.getChildren()) {
+                if (step.getPodInstanceRequirement().get().getPodInstance().getIndex() == replaceIndex) {
+                    continue;
+                }
+
+                PodInstanceRequirement restartPodInstanceRequirement =
+                        PodInstanceRequirement.newBuilder(
+                                step.getPodInstanceRequirement().get().getPodInstance(),
+                                step.getPodInstanceRequirement().get().getTasksToLaunch())
+                                .recoveryType(RecoveryType.TRANSIENT)
+                                .build();
+
+                restartSteps.add(
+                        new DefaultRecoveryStep(
+                                step.getName(),
+                                Status.PENDING,
+                                restartPodInstanceRequirement,
+                                new UnconstrainedLaunchConstrainer(),
+                                stateStore));
+            }
+        }
+
+        List<Step> steps = new ArrayList<>();
+        steps.add(replaceStep);
+        steps.addAll(restartSteps);
+
         return new DefaultPhase(
-                RECOVERY_PHASE_NAME, Arrays.asList(replaceStep), new SerialStrategy<>(), Collections.emptyList());
+                RECOVERY_PHASE_NAME,
+                steps,
+                new SerialStrategy<>(),
+                Collections.emptyList());
+    }
+
+    private boolean isSeedNode(int index) {
```

Comment:
nitpick: Not used anywhere yet, so no real reason to throw it in CassandraUtils. But also, seems like a CassandraUtils kinda joint.